features/cellular/framework/API/CellularNetwork.h

```diff
@@ -201,13 +200,38 @@ class CellularNetwork : public NetworkInterface
     };
     typedef CellularList<pdpcontext_params_t> pdpContextList_t;
 
+    /* Network registering mode */
+    enum NWRegisteringMode {
+        NWModeAutomatic = 0,    // automatic registering
+        NWModeManual,           // manual registering with plmn
+        NWModeDeRegister,       // deregister from network
+        NWModeSetOnly,          // set only <format> (for read command +COPS?), do not attempt registration/deregistration
+        NWModeManualAutomatic   // if manual fails, fallback to automatic
+    };
+
+
+    /** Does all the needed initializations that can fail
+     *
```

Comment:
Make a remark that this must be called immediately after constructor.

Reply:
check